### PR TITLE
fix(neon): use conventional macro syntax

### PR DESCRIPTION
**Pull request by ghost**
Since nobody seemed to be interested in my bugreport (https://github.com/lvgl/lvgl/issues/6647, which was also closed in the meantime), I'm proposing my patch as solution to the problem.

As explained in the bugreport the generated machine code is the same, only difference is that the assembler now just creates the symbols it's supposed to create.

The only downside is that the syntax is now a bit more verbose.

With best wishes,
Tobias

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` (`astyle v3.4.12` needs to installed by running `cd scripts; ./install_astyle.sh`) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Review by PeterBee97 — APPROVED** [review]
Sorry I've been busy with new job and don't really have time to test and fix things...Looks like the altmacro is all that's to blame? In that case as long as the problem is fixed, I'm fine with the verbose grammar. Thanks for the fix!

**Comment by tobiasjakobi-compleo** [comment]
@PeterBee97 Thanks!

Also I have tested our entire stack on real hardware yesterday. Visual inspection looks good, so at least from my side I can't spot any regression due to the macro syntax changes.

**Review by PeterBee97 — APPROVED** [review]

**Comment by tobiasjakobi-compleo** [comment]
Anything else I can do here to get this merged?

**Comment by PeterBee97** [comment]
@kisvegabor @FASTSHIFT Maybe help review and approve?

**Review by FASTSHIFT — APPROVED** [review]

File: src/draw/sw/blend/neon/lv_blend_neon.S [hunk 1]
```diff
@@ -70,249 +69,263 @@ TMP_Q1      .qn  q14
...
-        vsri.8          reg&_B, reg&_B, #5      @ BBBbbBBB
+        vshr.u8         TMP_Q0, \reg\()_565, #3    @ 000RRRrr 000gggBB
+        vshrn.i16       \reg\()_G, \reg\()_565, #5    @ rrGGGggg
+        vshrn.i16       \reg\()_R, TMP_Q0, #5      @ RRRrr000
+        vshl.i8         \reg\()_G, \reg\()_G, #2      @ GGGggg00
+        vshl.i16        TMP_Q1, \reg\()_565, #3    @ rrGGGggg BBBbb000
+        vsri.8          \reg\()_R, \reg\()_R, #5      @ RRRrrRRR
+        vmovn.i16       \reg\()_B, TMP_Q1          @ BBBbb000
+        vsri.8          \reg\()_G, \reg\()_G, #6      @ GGGgggGG
+        vsri.8          \reg\()_B, \reg\()_B, #5      @ BBBbbBBB
```
Comment (FASTSHIFT) [inline src/draw/sw/blend/neon/lv_blend_neon.S]: Please align code comments

**Review by FASTSHIFT — CHANGES_REQUESTED** [review]

**Comment by kisvegabor** [comment]
@EDGEMTech-GabrielC will try it out soon.

File: src/draw/sw/blend/neon/lv_blend_neon.S (hunk 1 above)
Comment (ghost) [reply src/draw/sw/blend/neon/lv_blend_neon.S]: Fixed. I've also aligned instruction arguments in the big `convert` macro. Should be much easier to read now.

**Review by FASTSHIFT — APPROVED** [review]

**Comment by kissa96** [comment]
LGTM! Thank you for the update Tobias!

**Comment by kisvegabor** [comment]
@EDGEMTech-GabrielC just test it and it worked, however we could measure a significant speed up. 🙁

**Review by kisvegabor — APPROVED** [review]

**Comment by liamHowatt** [comment]
> however we could measure a significant speed up. 🙁

My understanding is that these changes should not have a direct impact on performance.

**Comment by tobiasjakobi-compleo** [comment]
> > however we could measure a significant speed up. 🙁
> 
> My understanding is that these changes should not have a direct impact on performance.

The changes have neither direct not indirect impact here. Like I said above, the generated machine code is exactly the same.

**Comment by kisvegabor** [comment]
Clear, I just wanted point out what we have so far. 🙂

**Pull request merged**